File: OSCommon/src/org/optimizationservices/oscommon/datastructure/osentity/OptimizationInfo.java
/**
 * @(#)OptimizationInfo 1.0 03/14/2004
 *
 * Copyright (c) 2004
 */
package org.optimizationservices.oscommon.datastructure.osentity;


/**
 *
 * <p>A data structure class that corresponds to an xml element in the OSeL schema.  
 *  
 * </p>
 * @author Robert Fourer, Jun Ma, Kipp Martin
 * @version 1.0, 03/14/2004
 * @since OS 1.0
*/
public class OptimizationInfo{
		
	public String description = "";

	public String variableType = "";

	public String objectiveType = "";

	public String constraintType = "";

	public String objectiveLinearity = "";

	public String constraintLinearity = "";

	public String objectiveDifferentiability = "";

	public String constraintDifferentiability = "";

	public String parameterType = "";

	public String stochasticity = "";

	public boolean realTime = false;
	
	public FunctionType functionType = null;

	public SpecialStructure specialStructure = null;

	public SpecialAlgorithm specialAlgorithm = null;

}//class OptimizationInfo

File: OSCommon/org/optimizationservices/oscommon/datastructure/osentity/OptimizationInfo.cs
using System;

namespace org.optimizationservices.oscommon.datastructure.osentity{
	/// <summary>
	/// A data structure class that corresponds to an xml element in the OSeL schema. 
	/// @author Jun Ma	
	/// @version 1.0, 09/01/2005
	/// @since OS 1.0
	/// @copyright (c) 2005
	/// </summary>
	public class OptimizationInfo{

		public string description = "";

		public string variableType = "";

		public string objectiveType = "";

		public string constraintType = "";

		public string objectiveLinearity = "";

		public string constraintLinearity = "";

		public string objectiveDifferentiability = "";

		public string constraintDifferentiability = "";

		public string parameterType = "";

		public string stochasticity = "";

		public bool realTime = false;
	
		public FunctionType functionType = null;

		public SpecialStructure specialStructure = null;

		public SpecialAlgorithm specialAlgorithm = null;

	}//class OptimizationInfo
}//namespace
